csxtools/utils.py

```diff
@@ -128,6 +127,93 @@ def get_fastccd_images(light_header, dark_headers=None,
     return _correct_fccd_images(events, bgnd, flat, gain)
 
 
+def get_axis_images(light_header, dark_header=None, flat=None, tag=None, roi=None):
+    """Retreive and correct AXIS1 Images from associated headers
+
+    Retrieve AXIS1 Images from databroker and correct for:
+
+    -   Bad Pixels (converted to ``np.nan``)
+    -   Backgorund.
+    -   Flatfield correction.
+    -   Rotation (returned images are rotated 90 deg cw)
+
+    Parameters
+    ----------
+    light_header : databorker header
+        This header defines the images to convert
+
+    dark_headers : databroker headers , optional
+        The header is the dark images.
+
+    flat : array_like
+        Array to use for the flatfield correction. This should be a 2D
+        array sized as the last two dimensions of the image stack.
+
+
+    tag : string
+        Data tag used to retrieve images. Used in the call to
+        ``databroker.get_images()``. If `None`, use the defualt from
+        the settings.
+
+    roi : tuple
+        coordinates of the upper-left corner and width and height of
+        the ROI: e.g., (x, y, w, h)
+
+    Returns
+    -------
+    dask.array : corrected images
+
+    """
+    flipped_image = _get_axis1_images(light_header, dark_header, flat, tag, roi)
+    return flipped_image[..., ::-1]
+
+
+def _get_axis1_images(light_header, dark_header=None, flat=None, tag=None, roi=None):
+
+    if tag is None:
+        tag = detectors["axis1"]
```

Comment:
where does `detectors` come from?  If it comes from the scan header, then this is going to break.  A decision was made at the beamline to have 2 different modes of the axis (so detectors named `axis_standard` and `axis_cont`). Plus we will eventually use 2 different detectors interchangeably (`axis_cont` or `axis_standard` as we do now, AND `axis2_cont` or `axis2_standard` - which is the flange mounted detector).

I would suggest inspecting the the detectors list in the header for anything starting with axis.  If it starts withs `axis`, then we just append use that.  Please confirm that somewhere else in the code, `tag+"_image"`.  We could then confirm that this works on a 2 different scans when `tag` is not provided in the function call.

Reply:
Resolve as per https://github.com/NSLS-II-CSX/csxtools/issues/100 
Code is conservative to prevent unwanted KeyErrors